src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String DELIMITER = ",";
+    private static final String FIRST_WORD = "type";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_OF_FRUITS_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : inputReport) {
+            if (line.startsWith(FIRST_WORD)) {
+                continue;
+            }
+
+            String[] lineParts = line.split(DELIMITER);
+            validateLine(lineParts);
+            FruitTransaction transaction = new FruitTransaction();
+            transaction.setOperation(lineParts[OPERATION_TYPE_INDEX]);
+            transaction.setFruit(lineParts[FRUIT_NAME_INDEX]);
+            transaction.setQuantity(Integer.parseInt(lineParts[QUANTITY_OF_FRUITS_INDEX]));
+            transactions.add(transaction);
+        }
+        return transactions;
+    }
+
+    private void validateLine(String[] lineParts) {
```

Comment:
The `validateLine` method is missing access modifiers. According to the encapsulation principle, it's a good practice to explicitly declare the access level even for private methods.